### PR TITLE
Update bevy to 0.11

### PR DESCRIPTION
**Pull request by Olle-Lukowski**
Just updates bevy to 0.11, and changes some stuff to fix errors. Let me know if I should make any changes!

### CODE REVIEW

File: bevy_renet/src/transport.rs [hunk 1]
```diff
@@ -22,18 +22,15 @@ impl Plugin for NetcodeServerPlugin {
...
-        app.add_system(Self::update_system.in_base_set(CoreSet::PreUpdate).in_set(TransportSet::Server));
-        app.add_system(Self::send_packets.in_base_set(CoreSet::PostUpdate).in_set(TransportSet::Server));
-        app.add_system(
-            Self::disconnect_on_exit
-                .in_base_set(CoreSet::PostUpdate)
-                .in_set(TransportSet::Server),
-        );
+        app.add_systems(PreUpdate, Self::update_system.in_set(TransportSet::Server));
+        app.add_systems(PostUpdate, Self::send_packets.in_set(TransportSet::Server));
+        app.add_systems(PostUpdate, Self::disconnect_on_exit.in_set(TransportSet::Server));
```
Comment (Shatur) [inline bevy_renet/src/transport.rs:33]: Maybe group these systems in one call?
Similar for other places.

Comment (Olle-Lukowski) [reply bevy_renet/src/transport.rs]: Done!

File: bevy_renet/src/transport.rs [hunk 2]
```diff
@@ -65,18 +64,15 @@ impl Plugin for NetcodeClientPlugin {
...
-        app.add_system(Self::update_system.in_base_set(CoreSet::PreUpdate).in_set(TransportSet::Client));
-        app.add_system(Self::send_packets.in_base_set(CoreSet::PostUpdate).in_set(TransportSet::Client));
-        app.add_system(
-            Self::disconnect_on_exit
-                .in_base_set(CoreSet::PostUpdate)
-                .in_set(TransportSet::Client),
-        );
+        app.add_systems(PreUpdate, Self::update_system.in_set(TransportSet::Client));
+        app.add_systems(PostUpdate, Self::send_packets.in_set(TransportSet::Client));
+        app.add_systems(PostUpdate, Self::disconnect_on_exit.in_set(TransportSet::Client));
```
Comment (Shatur) [inline bevy_renet/src/transport.rs:75]: Could you combine these two systems too?

**Review by Shatur — APPROVED** [review]

**Comment by starwolfy** [comment]
There might be a regression. As the server I am unable to obtain the IP addresses of incoming connections. `NetcodeServerTransport` misses an implementation to return this data.

File: bevy_renet/src/lib.rs [hunk 3]
```diff
@@ -24,9 +24,9 @@ impl Plugin for RenetServerPlugin {
     fn build(&self, app: &mut App) {
         app.init_resource::<Events<ServerEvent>>();
 
-        app.configure_set(RenetSet::Server.run_if(resource_exists::<RenetServer>()));
+        app.configure_set(Update, RenetSet::Server.run_if(resource_exists::<RenetServer>()));
```
Comment (Shatur) [inline bevy_renet/src/lib.rs:27]: You configure set for Update, but use it in `PreUpdate`. I think that's the issue.

**Review by Shatur — CHANGES_REQUESTED** [review]

**Comment by lucaspoffo** [comment]
> There might be a regression. As the server I am unable to obtain the IP addresses of incoming connections. `NetcodeServerTransport` misses an implementation to return this data.

This is actually not exposed in the `NetcodeServerTransport`, it should be. I'll add it in another commit, pretty easy to do.

File: bevy_renet/src/lib.rs (hunk 3 above)
Comment (lucaspoffo) [reply bevy_renet/src/lib.rs:27]: Do we need to configure this with all default sets?
Like:
```rust
app.configure_set(PreUpdate, RenetSet::Server.run_if(resource_exists::<RenetServer>()));
app.configure_set(Update, RenetSet::Server.run_if(resource_exists::<RenetServer>()));
app.configure_set(PostUpdate, RenetSet::Server.run_if(resource_exists::<RenetServer>()));
```

This set is used to simplify the `run_if(resource_exists::<RenetServer>()`

Comment (lucaspoffo) [reply bevy_renet/src/lib.rs:27]: I guess we hit this issue:
https://github.com/bevyengine/bevy/issues/9139

Comment (lucaspoffo) [reply bevy_renet/src/lib.rs:27]: It seems we shouldn't be using SystemSets, and should just use directly `run_if(resource_exists::<RenetServer>` / `run_if(resource_exists::<RenetClient>`

Comment (Shatur) [reply bevy_renet/src/lib.rs:27]: I probably should remove the set enum too then.

**Review by lucaspoffo — APPROVED** [review]
Thanks! :pray: 
Will work on porting the examples, prob tomorrow/weekend will release the new version :+1:

**Comment by Shatur** [comment]
@lucaspoffo @Olle-Lukowski we forgot to remove `RenetSet` enum. And all `after` should be replaced with chain.

**Pull request merged**